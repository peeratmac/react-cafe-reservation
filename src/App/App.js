import React, { Component } from 'react';
import './App.css';
import ReservationsContainer from '../ReservationsContainer/ReservationsContainer';
import FormInput from '../FormInput/FormInput';

class App extends Component {
  constructor() {
    super();
    this.state = {
      reservations: []
    };
  }

  fetchExistingReservations() {
    fetch('http://localhost:3001/api/v1/reservations')
      .then(response => response.json())
      .then(data => this.setState({ reservations: data }));
  }

  componentDidMount() {
    this.fetchExistingReservations();
  }

  fetchWithPost = dataFromFormInput => {
    fetch('http://localhost:3001/api/v1/reservations', {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json'
      },
      body: JSON.stringify(dataFromFormInput)
    })
      .then(response => response.json())
      .then(data =>
        this.setState({ reservations: [...this.state.reservations, data] })
      )
      .catch(error => console.log(error));
  };

  // ! deprecated with the use of fetchWithPost function above
  // Todo: write new test
  addReservation = newReservation => {
    this.setState({
      reservations: [...this.state.reservations, newReservation]
    });
  };

  render() {
    console.log(this.state);
    return (
      <div className='App'>
        <h1 className='app-title'>Turing Cafe Reservations</h1>
        <FormInput addReservation={this.fetchWithPost} />
        <div className='resy-form'></div>
        <div className='resy-container'>
          <ReservationsContainer reservations={this.state.reservations} />
        </div>
      </div>
    );
  }
}

export default App;
