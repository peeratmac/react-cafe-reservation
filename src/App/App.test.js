import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import { shallow } from 'enzyme';

describe('APP', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<App />);
  });

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('should match App Snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should update state with a reservation when addReservation is called', () => {
    // Setup
    const mockReservation = {
      name: 'Joshua',
      date: '6/5',
      time: '12:00',
      number: 10
    };
    const expected = [mockReservation];

    // Expectation
    expect(wrapper.state('reservations')).toEqual([]);

    // Execution
    wrapper.instance().addReservation(mockReservation);

    // Expectaion
    expect(wrapper.state('reservations')).toEqual(expected);
  });
});
