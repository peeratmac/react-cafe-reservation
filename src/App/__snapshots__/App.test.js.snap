// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`APP should match App Snapshot 1`] = `
<div
  className="App"
>
  <h1
    className="app-title"
  >
    Turing Cafe Reservations
  </h1>
  <FormInput
    addReservation={[Function]}
  />
  <div
    className="resy-form"
  />
  <div
    className="resy-container"
  >
    <ReservationsContainer
      reservations={Array []}
    />
  </div>
</div>
`;
